src/chi-vue/src/components/data-table-save-view/SaveView.tsx

```diff
@@ -10,33 +10,32 @@ import {
   SAVE_VIEW_CLASSES,
 } from '@/constants/classes';
 import { SAVE_VIEW_EVENTS } from '@/constants/events';
-import { SaveViewModes } from '@/constants/types';
+import { SaveViewConfig, SaveViewModes } from '@/constants/types';
 import Tooltip from '../tooltip/tooltip';
 import { uuid4 } from '@/utils/utils';
 import './save-view.scss';
+import { defaultConfig } from './default-config';
 
 @Component
 export default class SaveView extends Vue {
-  @Prop() active!: boolean;
-  @Prop() results!: string;
-  @Prop() default?: boolean;
-  @Prop() id?: string;
-  @Prop() isPopoverActive?: boolean;
-  @Prop() mode?: SaveViewModes;
-  @Prop() title?: string;
+  @Prop() config!: SaveViewConfig;
 
-  isRibbonShown = !!this.$props.active;
-  isSaveViewVisible = !!this.$props.active;
-  isInfoPopoverActive = !!this.$props.isInfoPopoverActive;
-  isDefaultChecked = !!this.$props.default;
-  viewMode: SaveViewModes = this.$props.mode || SaveViewModes.BASE;
-  viewTitle = this.$props.title || '';
-  uuid: string | null = this.$props.id || null;
+  isRibbonShown = !!this.$props.config.active;
+  isSaveViewVisible = !!this.$props.config.active;
+  isDefaultChecked = !!this.$props.config.default;
+  viewMode: SaveViewModes = this.$props.config.mode || defaultConfig.mode;
+  viewTitle = this.$props.config.title || defaultConfig.title;
+  uuid: string | null = this.$props.config.id || null;
+  createModeLabel = this.$props.config.labels?.createLabel || defaultConfig.labels?.createLabel;
+  editModeLabel = this.$props.config.labels?.editLabel || defaultConfig.labels?.editLabel;
+  viewLabel = this.$props.config.labels?.viewLabel || defaultConfig.labels?.viewLabel;
+  defaultCheckboxLabel = this.$props.config.labels?.defaultCheckboxLabel || defaultConfig.labels?.defaultCheckboxLabel;
+  saveButtonDisabled = this.$props.config.saveButtonDisabled || defaultConfig.saveButtonDisabled;
 
-  @Watch('active')
-  updateVisibility(newValue: boolean, oldValue: boolean) {
+  @Watch('config')
+  dataConfigChange(newValue: SaveViewConfig, oldValue: SaveViewConfig) {
     if (newValue !== oldValue) {
-      this.isSaveViewVisible = newValue;
+      this.isSaveViewVisible = newValue.active;
```

Comment:
previously, `this.isSaveViewVisible` was changed only when prop `active` has changed.
Now it looks like, `this.isSaveViewVisible` will be updated when prop `config` has changed.

I suppose it would be better if we change condition to
` if (newValue.active !== oldValue.active) {`
So, `this.isSaveViewVisible` won't be changed if `config.active` remains the same, but some other config values were changed.